lib/sshutils/sshutils.go

```diff
@@ -0,0 +1,239 @@
+package sshutils
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/chef/automate/lib/logger"
+	"github.com/pkg/errors"
+	"golang.org/x/crypto/ssh"
+	"golang.org/x/crypto/ssh/knownhosts"
+)
+
+type SSHConfig struct {
+	SshUser    string
+	SshPort    string
+	SshKeyFile string
+	HostIP     string
+	Timeout    int
+}
+
+type SSHUtilImpl struct {
+	SshConfig *SSHConfig
+	SshClient ISshClient
+	logger    logger.Logger
+}
+
+type SSHUtil interface {
+	getClientConfig() (*ssh.ClientConfig, error)
+	SetSSHConfig(sshConfig *SSHConfig)
+	GetConnection() (*ssh.Client, error)
+	checkKnownHosts() (ssh.HostKeyCallback, error)
+	createKnownHosts() error
+	ConnectAndExecuteCommandOnRemoteWithSudoPassword(sshConfig *SSHConfig, sudoPassword string, sudoPasswordCmd string) (bool, error)
+	addHostKey(string, net.Addr, ssh.PublicKey) error
+}
+
+type ISshClient interface {
+	Dial(network string, addr string, config *ssh.ClientConfig) (*ssh.Client, error)
+	NewSession(*ssh.Client) (*ssh.Session, error)
+	ParsePrivateKey(pemBytes []byte) (ssh.Signer, error)
+	PublicKey(signers ssh.Signer) ssh.AuthMethod
+	CombinedOutput(cmd string, session *ssh.Session) ([]byte, error)
+	Close(*ssh.Session) error
+	Normalize(address string) string
+	New(files ...string) (ssh.HostKeyCallback, error)
+}
+
+type sshClient struct{}
+
+func NewSshClient() ISshClient {
+	return &sshClient{}
+}
+
+func (sc *sshClient) Dial(network string, addr string, config *ssh.ClientConfig) (*ssh.Client, error) {
+	conn, err := ssh.Dial(network, addr, config)
+	return conn, err
+}
+
+func (sc *sshClient) NewSession(conn *ssh.Client) (*ssh.Session, error) {
+	session, err := conn.NewSession()
+	return session, err
+}
+
+func (sc *sshClient) ParsePrivateKey(pemBytes []byte) (ssh.Signer, error) {
+	return ssh.ParsePrivateKey(pemBytes)
+}
```

Comment:
move all these to another file in same pkg > sshclient.go

Reply:
Done created the sshclient.go in the same package and mocksshclient.go